### PR TITLE
Add queue subscribe

### PR DESCRIPTION
**Pull request by piotrpio**
Signed-off-by: Piotr Piotrowski [piotr@synadia.com](mailto:piotr@synadia.com)

### CODE REVIEW

File: Sources/Nats/NatsSubscription.swift [hunk 1]
```diff
@@ -150,4 +160,26 @@ public class NatsSubscription: AsyncSequence {
...
+    // validateSubject will do a basic subject validation.
+    // Spaces are not allowed and all tokens should be > 0 in length.
+    private static func validSubject(_ subj: String) -> Bool {
+        let whitespaceCharacterSet = CharacterSet.whitespacesAndNewlines
+        if subj.rangeOfCharacter(from: whitespaceCharacterSet) != nil {
+            return false
+        }
+        let tokens = subj.split(separator: ".")
+        for token in tokens {
+            if token.isEmpty {
```
Comment (mtmk) [inline Sources/Nats/NatsSubscription.swift:173]: should we add wildcard checks? e.g. these tokens `abc*`, `>abc` and `a*bc` are invalid I think, no?

**Review by mtmk — APPROVED** [review]
LGTM

optional: we can add a bit more validation to subjects if my assumption is correct. looked easy enough I thought.

File: Sources/Nats/NatsSubscription.swift (hunk 1 above)
Comment (piotrpio) [reply Sources/Nats/NatsSubscription.swift:173]: I don't agree actually. It is legal to send wildcards within a subject name, even if it's not recommended. I checked with CLI, and you can normally publish ad subscribe to the subjects you mentioned above.

Additionally, we should not do extensive validation of subjects in the client on hotpaths (I treat core NATS pub/sub as such), as long as those subjects don't break server protocol (e.g. `\n` needs to be checked).

**Review by Jarema — APPROVED** [review]
LGTM

File: Sources/Nats/NatsSubscription.swift (hunk 1 above)
Comment (mtmk) [reply Sources/Nats/NatsSubscription.swift:173]: > It is legal to send wildcards within a subject name, even if it's not recommended. I checked with CLI, and you can normally publish ad subscribe to the subjects you mentioned above.

makes sense. thanks.

**Pull request merged**